File: rosetta/app/persistence/domain/record_file.go
// SPDX-License-Identifier: Apache-2.0

package domain

const tableNameRecordFile = "record_file"

type RecordFile struct {
	ConsensusStart   int64
	ConsensusEnd     int64 `gorm:"primaryKey"`
	Count            int64
	DigestAlgorithm  int
	FileHash         string
	HapiVersionMajor int
	HapiVersionMinor int
	HapiVersionPatch int
	Hash             string
	Index            int64
	LoadEnd          int64
	LoadStart        int64
	Name             string
	NodeId           int64
	PrevHash         string
	Version          int
}

// TableName returns record file table name
func (RecordFile) TableName() string {
	return tableNameRecordFile
}

File: rest/model/recordFile.js
// SPDX-License-Identifier: Apache-2.0

import camelCase from 'lodash/camelCase';
import mapKeys from 'lodash/mapKeys';

class RecordFile {
  /**
   * Parses record_file table columns into object
   */
  constructor(recordFile) {
    Object.assign(
      this,
      mapKeys(recordFile, (v, k) => camelCase(k))
    );
  }

  static tableAlias = 'rf';
  static tableName = 'record_file';

  static BYTES = 'bytes';
  static CONSENSUS_END = 'consensus_end';
  static CONSENSUS_START = 'consensus_start';
  static COUNT = 'count';
  static DIGEST_ALGORITHM = 'digest_algorithm';
  static FILE_HASH = 'file_hash';
  static GAS_USED = 'gas_used';
  static INDEX = 'index';
  static HAPI_VERSION_MAJOR = 'hapi_version_major';
  static HAPI_VERSION_MINOR = 'hapi_version_minor';
  static HAPI_VERSION_PATCH = 'hapi_version_patch';
  static HASH = 'hash';
  static LOAD_END = 'load_end';
  static LOAD_START = 'load_start';
  static LOGS_BLOOM = 'logs_bloom';
  static NAME = 'name';
  static PREV_HASH = 'prev_hash';
  static SIZE = 'size';
  static VERSION = 'version';

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }

  getFullHapiVersion() {
    return this.hapiVersionMajor === null || this.hapiVersionMinor === null || this.hapiVersionPatch === null
      ? null
      : `${this.hapiVersionMajor}.${this.hapiVersionMinor}.${this.hapiVersionPatch}`;
  }
}

export default RecordFile;
